app/scenes/mfa/mfa.js

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {Component} from 'react';
+import {injectIntl, intlShape} from 'react-intl';
+import {TextInput, Image, KeyboardAvoidingView} from 'react-native';
+
+import Button from 'react-native-button';
+import ErrorText from 'app/components/error_text';
+import FormattedText from 'app/components/formatted_text';
+import {GlobalStyles} from 'app/styles';
+
+import logo from 'assets/images/logo.png';
+
+import RequestStatus from 'service/constants/request_status';
+
+class Mfa extends Component {
+    static propTypes = {
+        actions: React.PropTypes.shape({
+            goBack: React.PropTypes.func.isRequired,
+            login: React.PropTypes.func.isRequired
+        }).isRequired,
+        intl: intlShape.isRequired,
+        loginId: React.PropTypes.string.isRequired,
+        password: React.PropTypes.string.isRequired,
+        loginRequest: React.PropTypes.object.isRequired
+    };
+
+    componentWillMount() {
+        this.setState({
+            token: '',
+            error: null
+        });
+    }
+
+    componentWillReceiveProps(nextProps) {
+        // In case the login is successful the previous scene (login) will take care of the transition
+        if (this.props.loginRequest.status === RequestStatus.STARTED &&
+            nextProps.loginRequest.status === RequestStatus.FAILURE) {
```

Comment:
The login screen is still mounted and in the stack so the `componentWillReceiveProps` is being invoked when we call the login action thus it will do what is suppose to do